src/rest/handlers/rest_events.erl

```diff
@@ -0,0 +1,131 @@
+%%%-------------------------------------------------------------------
+%% @doc
+%%  @todo
+%%  Implements API for handling events sent by other zones
+%% @end
+%%%-------------------------------------------------------------------
+
+-module(rest_events).
+-behavior(cowboy_handler).
+
+%% API
+-export([
+    init/2,
+    allowed_methods/2,
+    content_types_accepted/2,
+    content_types_provided/2,
+    resource_exists/2
+]).
+
+-export([
+    from_json/2,
+    to_json/2
+]).
+
+
+%%%---------------------------
+%% cowboy_rest callbacks
+%%%---------------------------
+
+init(Req0, State) ->
+    Method = cowboy_req:method(Req0),
+    #{operation := Operation} = State,
+    {ParsedParams, Req} = case {Method, Operation} of
+                              {<<"POST">>, single_event} ->
+                                  Params = cowboy_req:match_qs([{id, nonempty}], Req0),
+                                  {ok, Data, Req1} = cowboy_req:read_body(Req0),
+                                  Event = gmm_utils:decode(Data),
+                                  ok = validate_event(Event),
+                                  {maps:merge(Params, #{body => Event}), Req1};
+                              {<<"POST">>, bulk} ->
+                                  {ok, Data, Req1} = cowboy_req:read_body(Req0),
+                                  BulkEvents = gmm_utils:decode(Data),
+                                  ok = validate_bulk_events(BulkEvents),
+                                  #{<<"messages">> := MessagesList} = BulkEvents,
+                                  {#{body => MessagesList}, Req1};
+                              {<<"GET">>, stats} ->
+                                  {#{}, Req0}
+                          end,
```

Comment:
Wydaje mi się że przeniesienie case'a do nowej linijki poprawiło by czytelność. Trochę długie te linie i trochę dużo pustego miejsca. To też steamroll?

BTW, on zaraz zalicza wylot

Reply:
Spoko, się przeniesie, słuszna uwaga